src/main.cpp

```diff
@@ -4780,19 +4730,32 @@ bool ProcessBlock(CNode* pfrom, CBlock* pblock, bool generated_by_me)
     if (!mapBlockIndex.count(pblock->hashPrevBlock))
     {
         // *****      This area covers Gridcoin Orphan Handling      ***** 
-        if (true)
+        if (WalletOutOfSync())
         {
-            if (WalletOutOfSync())
+            if (TimerMain("OrphanBarrage",100))
             {
-                if (TimerMain("OrphanBarrage",100))
+                // If we stay out of sync for more than 25 orphans and never recover without accepting a block - attempt to recover the node- if we recover, reset the counters.
+                // We reset these counters every time a block is accepted successfully in AcceptBlock().
+                // Note: This code will never actually be exercised unless the wallet stays out of sync for a very long time - approx. 24 hours - the wallet normally recovers on its own without this code.
+                // I'm leaving this in for people who may be on vacation for a long time - it may keep an external node running when everything else fails.
+                if (TimerMain("CheckForFutileSync", 25))
                 {
-                    mapAlreadyAskedFor.clear();
-                    printf("\r\nClearing mapAlreadyAskedFor.\r\n");
-                    AskForOutstandingBlocks(uint256(0));
-                    CheckForFutileSync();
+                    mapOrphanBlocks.clear();
```

Comment:
why clear this map?

Reply:
Tbh I just moved the contents of `CheckForFutileSync` down here. I don't know if this needs be be refactored further.

Reply:
It looks shady to me. But unrelated to this change. I Still approve.

Reply:
The map clearing will leak memory, switch to `clearOrphanBlocks()` when https://github.com/gridcoin/Gridcoin-Research/pull/609 is merged ;-)